IMS.BL/Inventory.cs

```diff
@@ -1,52 +1,44 @@
-using System.Collections.Generic;
-using System.Linq;
-using AutoMapper;
 using System;
+using System.Threading.Tasks;
+using IMS.BL.DataService;
 
 namespace IMS.BL
 {
     public class Inventory
     {
-        public Inventory()
+        private readonly DataAccess _dataAccess;
+        public Inventory(DataAccess dataAccess)
         {
-            ProductsList = new Dictionary<string, Product>();
+            _dataAccess = dataAccess;
         }
 
-        public  Dictionary<string, Product> ProductsList { get; private set; }
-
-        public  Product AddNewProduct(ProductData productData, Mapper mapper)
+        public  async Task<bool> AddNewProduct(ProductData productData)
         {
-            var product = mapper.Map<ProductData, Product>(productData);
-            
-            ProductsList.Add(productData.ProductName, product);
-
-            return product;
+            return await _dataAccess.InsetProduct(productData);
         }
 
-        public void EditProductByName(string productName, ProductData productData)
+        public async Task<bool> EditProductByName(string productId, ProductData productData)
         {
-            var product = ProductsList.Values.SingleOrDefault(prod => prod.ProductName == productName);
-
-            if (product == null)
+            var isUpdated = await _dataAccess.UpdateProduct(productId, productData);
+            
+            if (!isUpdated)
             {
                 throw new NullReferenceException("No products can be found.");
             }
-            
-            product.ProductName = productData.ProductName;
-            product.ProductPrice = productData.ProductPrice;
-            product.ProductQuantity = productData.ProductQuantity;
+
+            return isUpdated;
         }
 
-        public void RemoveProductByName(string productName)
+        public async Task<bool> RemoveProduct(string productId)
```

Comment:
why you are unnconsistent ? on the SQLProvider this method was bool? and you did not use synchronization with it, thats why what you should have don first if refactor your whole system and apply the new concepts that you have acquire like strategy pattern and other topics then the 2 different database providers.